lib/ansible/modules/cloud/azure/azure_rm_acr.py

```diff
@@ -0,0 +1,362 @@
+        provisioning_state=registry.provisioning_state if registry is not None else "",
+        password_name=credentials.passwords[0].name.value if credentials is not None else "",
+        password_value=credentials.passwords[0].value if credentials is not None else "",
+        password2_name=credentials.passwords[1].name.value if credentials is not None else "",
+        password2_value=credentials.passwords[1].value if credentials is not None else "",
+        tags=registry.tags if registry is not None else ""
+    )
+    return results
+
+
+class Actions:
+    NoAction, Create, Update = range(3)
+
+
+class AzureRMContainerRegistry(AzureRMModuleBase):
+    """Configuration class for an Azure RM container registry resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            admin_user_enabled=dict(
+                type='bool',
+                required=False,
+                default=False
+            ),
+            sku=dict(
+                type='str',
+                required=False,
+                default='Basic',
+                choices=['Basic', 'Standard', 'Premium']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.state = None
+        self.sku = None
+        self.tags = None
+        self._containerregistry_mgmt_client = None
+
+        self.results = dict(changed=False, state=dict())
+
+        super(AzureRMContainerRegistry, self).__init__(
+            derived_arg_spec=self.module_arg_spec,
+            supports_check_mode=True,
+            supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        resource_group = None
+        response = None
+        to_do = Actions.NoAction
+
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            self.location = resource_group.location
+
+        # Check if the ACR instance already present in the RG
+        if self.state == 'present':
+            response = self.get_acr()
+            self.results['state'] = response
+
+            if not response:
+                to_do = Actions.Create
+            else:
+                self.log('Results : {0}'.format(response))
+                if response['provisioning_state'] == "Succeeded":
```

Comment:
This doesn't appear to support proper idempotence; it looks like this module will always report `changed: true` since it doesn't inspect the state of the existing registry against the requested state from the module args before determining that an update is necessary.